upstairs/src/lib.rs

```diff
@@ -6986,6 +6916,16 @@ impl BlockIO for Guest {
         }
     }
 
+    async fn deactivate(&self) -> Result<BlockReqWaiter, CrucibleError> {
+        // Disable any more IO from this guest and deactivate the downstairs.
+        // We can't deactivate if we are not yet active.
+        if !self.is_active().await {
```

Comment:
This is checking  the `Guest` local `is_active`, right?
I worry that the guest level `is_active` can be out of sync.
Can we just let the IO through here and if the Upstairs is not active have it return the error?

It may mean some more logic changing around this, and maybe candidate for a follow up PR?

Reply:
I agree. I opened https://github.com/oxidecomputer/crucible/issues/461 to track this